web/js/layers/wv.layers.modal.js

```diff
@@ -120,458 +115,406 @@ wv.layers.modal = wv.layers.modal || function(models, ui, config) {
+        var sortNumber;
+
+        // Check if categories have settings with the same projection.
+        var categoryHasSetting;
+        _.each( category.measurements, function( measurement, index ) {
+          var projection = models.proj.selected.id;
+          var current = config.measurements[measurement];
+          _.each( current.sources, function( source, sourceName ) {
+            _.each( source.settings, function( setting ) {
+              var layer = config.layers[setting];
+              var proj = layer.projections;
+              if (layer.layergroup && layer.layergroup.indexOf("reference_orbits") !== -1) {
+                if(current.id === "orbital-track") {
+                  categoryHasSetting = true;
+                }
+                // Don't output categories with only orbits
+              } else {
+                categoryHasSetting = true;
+              }
+            });
+          });
+        });
 
-      $('label.search-icon')
-        .addClass('search-on');
-      $('label.search-on')
-        .click(showDefaultPage);
+        if (categoryHasSetting === true) {
+          if (category.placement) {
+            if (category.placement === 'first') {
+              sortNumber = 1;
+            } else if (category.placement === 'last') {
+              sortNumber = 3;
+            }
+          } else {
+            sortNumber = 2;
+          }
+          var $category = $('<div></div>').attr('data-sort', sortNumber).addClass('layer-category layer-category-' + interestCssName(metaCategoryName)).attr('id', category.id);
+          if (category.image) {
+            $category.css('background-image', 'url("images/wv.layers/categories/' + category.image + '")');
+          }
 
-      $breadcrumb.show();
-    }
-  };
+          var $categoryOpaque = $('<div></div>').addClass('category-background-cover');
 
-  var cssName = function(name) {
-    if (name === 'hazards and disasters') {
-      return 'legacy';
-    } else {
-      return name;
-    }
-  };
+          $category.append($categoryOpaque);
 
-  var replaceIfScientific = function(name) {
-    if (name === 'scientific') {
-      return 'science disciplines';
-    } else {
-      return name;
-    }
-  };
+          var $categoryTitle = $('<h3></h3>');
 
-  var setCategoryOverflow = function(category, $measurements) {
-    var $dotContinueItem = $('<li></li>')
-      .addClass('layer-category-item');
+          var $categoryLink = $('<a></a>').text(category.title).attr('alt', category.title).addClass('layer-category-name').click(function(e) {
+            drawMeasurements(category);
+          });
 
-    var $dotContinueLink = $('<a></a>')
-      .attr('data-category', category.id)
-      .text('...');
+          $categoryTitle.append($categoryLink);
+          $categoryOpaque.append($categoryTitle);
+
+          var $measurements = $('<ul></ul>');
+          $i = 0;
+          _.each(category.measurements, function(measurement, index) {
```

Comment:
@ZachTRice looks like a similar nested loop is being reused a few times. I think you could throw this into a function and reuse.

Reply:
I have updated the redundant part of the function which is just inside the code snippet you highlighted. That beginning `_.each` category loop is only used once but the source setting check inside is used multiple time which is what I changed.